### PR TITLE
diff: clean checkpoint info in several times to avoid `transaction is too large` error in TiDB

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
fix issue: https://internal.pingcap.net/jira/browse/TOOL-1432
tidb only support delete less than 300,000 line one transaction,

### What is changed and how it works?
1. delete 100,000 rows once
2. add check for chunk-size
3. update summary after checkpoint is loaded to avoid warn log

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: pkg/dbutil/common.go [hunk 1]
```diff
@@ -756,3 +756,22 @@ func ignoreDDLError(err error) bool {
 		return false
 	}
 }
+
+// DeleteRows delete rows in several times. Only can delete less than 300,000 one time in TiDB.
+func DeleteRows(ctx context.Context, db *sql.DB, schemaName string, tableName string, where string, args []interface{}) error {
```
Comment (july2993) [inline pkg/dbutil/common.go:761]: why not just truncate table

Comment (kennytm) [reply pkg/dbutil/common.go:761]: TRUNCATE doesn't accept a `where` clause

File: pkg/dbutil/common_test.go [hunk 2]
```diff
@@ -129,3 +131,19 @@ func (s *testDBSuite) TestIsIgnoreError(c *C) {
...
+	// delete and select twice
+	mock.ExpectExec("DELETE FROM").WillReturnResult(sqlmock.NewResult(0, 1))
+	countRows := sqlmock.NewRows([]string{"cnt"}).AddRow(199999)
+	mock.ExpectQuery("SELECT COUNT.*").WillReturnRows(countRows)
+	mock.ExpectExec("DELETE FROM").WillReturnResult(sqlmock.NewResult(0, 1))
+	countRows = sqlmock.NewRows([]string{"cnt"}).AddRow(0)
+	mock.ExpectQuery("SELECT COUNT.*").WillReturnRows(countRows)
+
+	err = DeleteRows(context.Background(), db, "test", "t", "", nil)
+	c.Assert(err, IsNil)
```
Comment (3pointer) [inline pkg/dbutil/common_test.go:144]:
```
if err := mock.ExpectationsWereMet(); err != nil {
		c.Errorf("there were unfulfilled expectations: %s", err)
}
```

File: pkg/dbutil/common.go [hunk 3]
```diff
@@ -756,3 +756,22 @@ func ignoreDDLError(err error) bool {
...
+	deleteSQL := fmt.Sprintf("DELETE FROM `%s`.`%s` WHERE %s limit 100000;", schemaName, tableName, where)
+	_, err := db.ExecContext(ctx, deleteSQL, args...)
```
Comment (july2993) [inline pkg/dbutil/common.go]: can we just check RowsAffected instead of `GetRowCount`

Comment (WangXiangUSTC) [reply pkg/dbutil/common.go]: https://github.com/pingcap/tidb-tools/pull/258/commits/bee3bb51f92908f70d171c6520124cdb5d2ab4db

File: pkg/dbutil/common.go [hunk 4]
```diff
@@ -45,6 +45,9 @@ const (
 
 	// SlowWarnLog defines the duration to log warn log of sql when exec time greater than
 	SlowWarnLog = 100 * time.Millisecond
+
+	// DefaultDeleteRowsNum is the default rows num for delete one time
+	DefaultDeleteRowsNum = 100000
```
Comment (IANTHEREAL) [inline pkg/dbutil/common.go]: is it reasonable?

File: pkg/dbutil/common.go [hunk 5]
```diff
@@ -161,7 +164,7 @@ func GetCreateTableSQL(ctx context.Context, db *sql.DB, schemaName string, table
 
 // GetRowCount returns row count of the table.
 // if not specify where condition, return total row count of the table.
-func GetRowCount(ctx context.Context, db *sql.DB, schemaName string, tableName string, where string) (int64, error) {
+func GetRowCount(ctx context.Context, db *sql.DB, schemaName string, tableName string, where string, args []interface{}) (int64, error) {
```
Comment (IANTHEREAL) [inline pkg/dbutil/common.go]: where to use `args`?

File: pkg/dbutil/common.go (hunk 4 above)
Comment (IANTHEREAL) [inline pkg/dbutil/common.go]:
```suggestion
	DefaultDeleteRowsNum int64 = 100000
```

File: pkg/diff/diff.go [hunk 6]
```diff
@@ -168,7 +171,12 @@ func (t *TableDiff) CheckTableStruct(ctx context.Context) (bool, error) {
 
 func (t *TableDiff) adjustConfig() {
 	if t.ChunkSize <= 0 {
-		t.ChunkSize = 100
+		log.Warn("chunk size is less than 0, will use default value 1000", zap.Int("chunk size", t.ChunkSize))
+		t.ChunkSize = 1000
+	}
+
+	if t.ChunkSize < 1000 || t.ChunkSize > 10000 {
+		log.Warn("chunk size is suggest in range [1000, 10000]", zap.Int("chunk size", t.ChunkSize))
```
Comment (IANTHEREAL) [inline pkg/diff/diff.go]:
```suggestion
		log.Warn("chunk size is recommend in range [1000, 10000]", zap.Int("chunk size", t.ChunkSize))
```

**Comment by IANTHEREAL** [comment]
Rest LGTM

File: pkg/dbutil/common.go (hunk 4 above)
Comment (WangXiangUSTC) [reply pkg/dbutil/common.go]: which number do you suggest?

File: pkg/dbutil/common.go (hunk 5 above)
Comment (WangXiangUSTC) [reply pkg/dbutil/common.go]: removed

**Comment by IANTHEREAL** [comment]
LGTM

**Review by july2993 — APPROVED** [review]
LGTM

**Pull request merged**